### ISSUE TITLE
Gulp Plugin Doesn't Show Compilation Errors

### ISSUE THREAD

**Issue by dsanders11** [google/closure-compiler-npm#122]
I've been getting a lot of compilation errors while trying to migrate to ES6 modules (some trial and error), and every time an error happens I get the following not very useful output:

`UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: `): [object Object]`

I've tracked it back to the gulp plugin, and if I add a `.catch(err => { console.log(err); })` to [this line](https://github.com/google/closure-compiler-npm/blob/master/packages/google-closure-compiler/lib/gulp/index.js#L215) I can get useful output. Tracing it further, I see that the error is correctly admitted from `_compilationComplete` (on [this line](https://github.com/google/closure-compiler-npm/blob/master/packages/google-closure-compiler/lib/gulp/index.js#L227)) but from there it is swallowed up and the unhelpful error is all I see.

**Comment by ChadKillingsworth** [comment]
Can you give me an example of the type of code that is causing these errors? I'd like to add a test for this as I fix it.

**Comment by dsanders11** [comment]
@ChadKillingsworth, one that I found which would cause it would be try to assign to a const variable, so:

```javascript
const foo = 3;
foo = 2;
```

**Comment by ChadKillingsworth** [comment]
It's going to take more than that. Normal errors are properly reported. Here's an example using your test case:

<img width="741" alt="screen shot 2018-12-10 at 6 48 17 am" src="https://user-images.githubusercontent.com/1247639/49733628-a2435780-fc47-11e8-80f8-1d8fea935832.png">

However, all promises should have a catch clause so I added one. I audited the code and I'm still not quite sure how that catch would ever be triggered though. I had to add code that would throw an exception.

**Comment by dsanders11** [comment]
@ChadKillingsworth, sorry I couldn't get a working reproduce case for you, but it looks like the change you made in #124 fixed the issue for me. 👍
